src/screens/navscreens/MyPickupScreen.js

```diff
@@ -1,12 +1,123 @@
 import React from 'react';
-import {StyleSheet, View, Text} from 'react-native';
+import { StyleSheet, View, Text, Image } from 'react-native';
 import NavScreenHeader from '../../components/general/NavScreenHeader';
+import { NavigationActions } from 'react-navigation';
+import { FlatList } from 'react-native-gesture-handler';
 
+const courierImgs = {
+  ups: 'https://cdn.iconscout.com/icon/free/png-512/ups-282281.png',
+  usps: 'https://cdn4.iconfinder.com/data/icons/logos-and-brands/512/353_Usps_logo-512.png',
+  fedex: 'https://cdn.iconscout.com/icon/free/png-512/fedex-3-283136.png',
+};
+
+const pickups = [
+  {
+    id: 1,
+    key: 'Asus Laptop',
+    date: 'Dec 17, 2019 2PM - 4PM',
+    status: 'Awaiting pickup',
+    courier: 'usps',
+    delivery: 1,
+    price: 4.00,
+    printing: 0.5,
+    packaging: 2.00,
+    tax: 0.75
+  },
+  {
+    id: 2,
+    key: 'Macbook Pro',
+    date: 'Dec 18, 2019 2PM - 4PM',
+    status: 'In transit',
+    courier: 'ups',
+    delivery: 1,
+    price: 4.00,
+    printing: 0.5,
+    packaging: 2.00,
+    tax: 0.75
+  },
+  {
+    id: 3,
+    key: 'Surfacebook',
+    date: 'Dec 19, 2019 2PM - 4PM',
+    status: 'Completed',
+    courier: 'fedex',
+    delivery: 1,
+    price: 4.00,
+    printing: 0.5,
+    packaging: 2.00,
+    tax: 0.75
+  },
+];
+
+const styles = StyleSheet.create({
+  pickupContainer: {
+    padding: 20
+  },
+  pickupItem: {
+    textAlign: 'right',
+    fontSize: 16
+  },
+  status: {
+    fontWeight: 'bold'
+  },
+  row: {
+    flexDirection: 'row',
+    paddingTop: 20,
+    justifyContent: 'space-between',
+    alignItems: 'center'
+  },
+  icon: {
+    width: 50,
+    height: 50,
+    justifyContent: 'center'
+  }
+})
 export default class MyPickupScreen extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  navigateToScreen = (route, item) => () => {
+    const navigateAction = NavigationActions.navigate({
+      routeName: route,
+      params: {
+        pickup: item
+      }
+    });
+    this.props.navigation.dispatch(navigateAction);
+  };
   render() {
+    console.log(pickups);
     return (
       <View>
-        <NavScreenHeader navigation={this.props.navigation} title={'My Pickups'}/>
+        <NavScreenHeader navigation={this.props.navigation} title={'My Pickups'} />
+        <View style={styles.pickupContainer}>
+          <FlatList
+            data={pickups}
+            renderItem={({ item }) =>
+              <View>
+                <View>
+                  <View key={item.key} onStartShouldSetResponder={this.navigateToScreen('Pickup', item)}>
```

Comment:
Instead of using onStartShouldSetResponder, can you wrap the view with a TouchableNativeFeedback and use the onPress on that TouchableNativeFeedback.  That way the user can see some feedback when they press on it and they know it's a clickable component. Instead of the dispatch/navigateToScreen method, you can just do this.props.navigation.navigate("Pickup",  {pickup: item})